src/Microsoft.ML.Tokenizers/Model/EnglishRobertaTokenizer.cs

```diff
@@ -597,14 +597,14 @@ private int EncodeToIdsResult(List<EncodedToken> tokens, IList<int>? accumulated
                 for (int i = 0; i < maxTokens; i++)
                 {
                     accumulatedIds.Add(tokens[i].Id);
-                    charsConsumed += tokens[i].Offset.Length;
+                    charsConsumed += tokens[i].Offset.End.Value - tokens[i].Offset.Start.Value;
```

Comment:
Now that we are using `Range` and can use the "from the end" notation, is taht something we need to worry about here or is it being controlled/created in such a way it will never use that notation?

Reply:
Good question, we always fill the exact index as the length of the container will not be available all the time. It is a requirement for this scenario to fill the range with exact offsets.